### PR TITLE
Make components macvtap

### PR DESCRIPTION
**Pull request by maiqueb**
Automate macvtap component generation.

By doing `make components`, the specified version of macvtap-cni is cloned, and it's manifests copied, becoming available as CNAO provisioning files.

Depends-on: #346 
Depends-on: https://github.com/kubevirt/macvtap-cni/pull/20

### CODE REVIEW

File: hack/components.sh [hunk 1]
```diff
@@ -71,3 +71,47 @@ KUBEMACPOOL_IMAGE=quay.io/kubevirt/kubemacpool
...
+
+echo 'Get macvtap-cni image name and update it under CNAO'
+MACVTAP_TAG=$(
+    cd ${MACVTAP_PATH}
+    git describe --tags
+)
+MACVTAP_IMAGE=quay.io/kubevirt/macvtap-cni
+MACVTAP_IMAGE_TAGGED=${MACVTAP_IMAGE}:${MACVTAP_TAG}
+sed -i "s#\"${MACVTAP_IMAGE}:.*\"#\"${MACVTAP_IMAGE_TAGGED}\"#" pkg/components/components.go
+sed -i "s#\"${MACVTAP_IMAGE}:.*\"#\"${MACVTAP_IMAGE_TAGGED}\"#" test/releases/${CNAO_VERSION}.go
```
Comment (phoracek) [inline hack/components.sh]: This is not needed **yet**. Only once we add upgrade support for macvtap, we add it to releases/ and there will be something to replace. I'm afraid we would forget about this, so please just comment it out and add a TODO (rare exception to have code commented out / it's too late)

File: components.yaml [hunk 2]
```diff
@@ -17,3 +17,6 @@ components:
   ovs-cni:
     url: "https://github.com/kubevirt/ovs-cni"
     commit: "23d572171e383009aefa693832d1d8a7b8d04b22" # v0.11.0
+  macvtap-cni:
+    url: "https://github.com/kubevirt/macvtap-cni"
+    commit: "0c1d111aab4b8222ac2b341d0ce53981649ba2a3" #v0.1.0
```
[the remark this answers is not in the document]
Comment (maiqueb) [reply components.yaml]: Hmm... I don't follow. Sorry.

File: hack/components.sh (hunk 1 above)
Comment (maiqueb) [reply hack/components.sh]: Done.

File: components.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (maiqueb) [reply components.yaml]: > Oh no, my OCD. Please fix it

Done.

File: hack/components.sh [hunk 3]
```diff
@@ -71,3 +71,48 @@ KUBEMACPOOL_IMAGE=quay.io/kubevirt/kubemacpool
...
+
+echo 'Get macvtap-cni image name and update it under CNAO'
+MACVTAP_TAG=$(
+    cd ${MACVTAP_PATH}
+    git describe --tags
+)
+# TODO: uncomment the following lines *once* there is macvtap upgrade is supported
+#MACVTAP_IMAGE=quay.io/kubevirt/macvtap-cni
+#MACVTAP_IMAGE_TAGGED=${MACVTAP_IMAGE}:${MACVTAP_TAG}
+#sed -i "s#\"${MACVTAP_IMAGE}:.*\"#\"${MACVTAP_IMAGE_TAGGED}\"#" pkg/components/components.go
```
Comment (phoracek) [inline hack/components.sh]: This should be still active. components.go bump the defaults (so image value used in the generated manifests). Only the following line is related to upgrade tests.

**Comment by phoracek** [comment]
/lgtm
/approve

Looking forward to test it with 0.2!

**Comment by phoracek** [comment]
/override pull-e2e-cluster-network-addons-operator-workflow-k8s

**Pull request merged**